pkg/gotohelm/transpiler.go

```diff
@@ -97,13 +101,21 @@ type Transpiler struct {
 	// namespaces is a cache for holding the transpiled name of a function.
 	// It's exclusively used by `funcNameFor`.
 	names map[*types.Func]string
+
+	pkgToProcess []*packages.Package
 }
 
 func (t *Transpiler) Transpile() *Chart {
 	var chart Chart
-	for _, f := range t.Files {
-		if transpiled := t.transpileFile(f); transpiled != nil {
-			chart.Files = append(chart.Files, transpiled)
+
+	for _, p := range t.pkgToProcess {
+		t.Package = p
```

Comment:
I'm a staunch believer that mutability is the root of evil. Rather than making the transpiler's internal state change depending on how it's called, move this logic "up" into the top level `Transpile` call. If this starts to get really hairy, it might make sense to make two different concepts, a `PackageTranspiler` and a `ChartTranspiler`.

We actually, have some of this already built already because of how the `bootstrap`/`shims` package works. Let's take that functionality and extend it to be more generic.

1. Modify `Chart` to have a `SubCharts` or `Dependencies` `[]*Chart` field. Have `Transpile` automatically include `shims` in that field (it's currently added in `Transpiler.Transpile` but we don't want that to be the case anymore).
2. Modify `Transpiler` to have a `subcharted []*packages.Package` field. (There may be a more efficient data structure. Up to you!). In `transpileCallExpr` extend the block that starts with `if callee.Pkg().Path() == t.Package.PkgPath` to instead check if a call is coming from a "subcharted" package and otherwise behave the same way.
3. Modify `Transpile` to roughly do:
```go
func Transpile(root *packages.Package, deps []*packages.Package) (*Chart, error) {
// We want root to be in deps so a chart calling it's own functions vs calling subcharted functions works the same way.
deps = append(deps, root)

chart := newTranspiler(root, deps)
for _, dep := range deps {
    chart.SubCharts = append(chart.SubCharts, newTranspiler(dep, deps))
}

// Always append the shims subchart (Moved up from the Transpiler.Transpile call)
chart.Subcharts = append(charts.Subcharts, shims)
}
```

And I think that's all it'll take to get this working as desired 🎉 we know that everything works already because the `shims` module (namespacing) works.